lib/handleRequest.js

```diff
@@ -126,7 +126,11 @@ function preHandlerCallback (err, code) {
         this.reply.send(payload)
       }
     }).catch((err) => {
-      this.reply.send(err)
+      if (err instanceof Error) {
+        this.reply.send(err)
+      } else {
+        this.reply.send(new Error(err || ''))
```

Comment:
I'm not not very happy about this part, can you do a check if `err`  is a string and use that with `new Error`, or log the object with a warning?

Reply:
Is there a reason why it must be a string? I think the Error constructor does this already. See: https://tc39.github.io/ecma262/#sec-error-message

FYI the error-like objects that I was getting from my app's dependency actually have a `toString` method so this fix works.

Reply:
I want to avoid:

```
> new Error({ hello: 'world' })
Error: [object Object]
    at repl:1:1
    at ContextifyScript.Script.runInThisContext (vm.js:50:33)
    at REPLServer.defaultEval (repl.js:240:29)
    at bound (domain.js:301:14)
    at REPLServer.runBound [as eval] (domain.js:314:12)
    at REPLServer.onLine (repl.js:441:10)
    at emitOne (events.js:121:20)
    at REPLServer.emit (events.js:211:7)
    at REPLServer.Interface._onLine (readline.js:282:10)
    at REPLServer.Interface._line (readline.js:631:8)
```

We can check if the object has a `toString` method that is not `Object.prototype.toString`.

Reply:
Actually I feel like this type checking and coercion is premature here. It should take place after handleError (and any `reply.context.errorHandler` hooks) have processed the raw error.

Reply:
@sebdeckers all of what you have mentioned assumes an error is an `Error`. Refactoring that would be a major undertaking, and I do not think it is worth it to support this edge case.